src/app/commands/cmd_paste.cpp

```diff
@@ -30,12 +33,17 @@ PasteCommand::PasteCommand()
 
 bool PasteCommand::onEnabled(Context* ctx)
 {
-  return App::instance()->inputChain().canPaste(ctx);
+  return ctx->isUIAvailable() ?
+    App::instance()->inputChain().canPaste(ctx) :
+    ctx->clipboard()->isImageAvailable();
 }
 
 void PasteCommand::onExecute(Context* ctx)
 {
-  App::instance()->inputChain().paste(ctx);
+  if (ctx->isUIAvailable())
+    App::instance()->inputChain().paste(ctx);
+  else if (ctx->clipboard())
+    ctx->clipboard()->paste(ctx, false);
```

Comment:
Probably we required a `origin` argument for the paste (at least initially) to paste the image in any position:

```c++
struct PasteParams : public NewParams {
  Param<gfx::Point> origin { this, gfx::Point(), "origin" };
};

class PasteCommand : public CommandWithNewParams<PasteParams> {
public:
  PasteCommand();
...
```